plugins/fr.cea.nabla.ir/src/fr/cea/nabla/ir/transformers/CreateArrayOperators.xtend

```diff
@@ -242,14 +241,8 @@ class CreateArrayOperators extends IrTransformationStep
 		block.instructions += IrFactory::eINSTANCE.createVariableDeclaration => [ variable = result ]
 
 		// loops and binary operation
-		val resultRef = opUtils.createArgOrVarRef(result)
 		val aRef = opUtils.createArgOrVarRef(a)
-		val Interval[] intervals = newArrayOfSize(variables.size)
-		for (i : 0..<intervals.size) intervals.set(i, createInterval(variables.get(i)))
-		block.instructions += opUtils.createLoopForUnaryOp(resultRef, aRef, intervals, op)
-
-		// return instruction to return the result
-		block.instructions += IrFactory::eINSTANCE.createReturn => [ expression = opUtils.createArgOrVarRef(result) ]
+		block.instructions += opUtils.createReturnForUnaryOp(aRef, op)
```

Comment:
These changes break all code generators but DaCe. Choose to either remove other code generators from this fork, or to refactor and duplicate code for DaCe and non-DaCe generators. This may involve heavy code refactoring.